### PR TITLE
Interactive Example

### PR DESCRIPTION
**Pull request by nils-braun**
This PR adds an "interactive" example to the tsfresh website.

image

It is possible to specify a CSV file and the column specifications by just clicking and typing and submitting the file to the backend, which will return the calculated features.

I tried to include what I have learned from you @dotcs - but maybe I failed horribly :-D

### CODE REVIEW

File: src/components/hero/index.tsx [hunk 1]
```diff
@@ -1,60 +1,50 @@
...
-            <br />
-            <br />
-            <span className="whitespace-no-wrap">
-              features = extract_relevant_features(timeseries, y, column_id=
-              <span className="text-blue-600">"id"</span>,
-              column_sort=
-              <span className="text-blue-600">"time"</span>)
-            </span>
-            <br />
+export default class Hero extends React.Component<{}, HeroState> {
```
Comment (dotcs) [inline src/components/hero/index.tsx]: Can be rewritten as a FunctionalComponent with the help of `useState`. This is now the preferred way when writing stateful components.

File: src/components/hero/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/// Fake downloading of a file by adding a <a></a>
+export function serveFile(blob : any, fileName : string = "result.csv") {
```
Comment (dotcs) [inline src/components/hero/utils.tsx:2]: Sometimes I really love Javascript ... NOT :sweat_smile: 
But it's indeed how things are done.

File: src/components/hero/interactive.tsx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+              onChange={this.handleChange}
+              value={this.state.column_sort}
+            />
+            <span>,</span>
+          </span>
+          <br />
+          <span className="whitespace-no-wrap">
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+            &nbsp;&nbsp;&nbsp;
```
Comment (dotcs) [inline src/components/hero/interactive.tsx]: This is ugly. Why is this needed?

File: src/components/hero/interactive.tsx [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  column_sort: string;
+  column_kind: string;
+  data_file: string;
+  loading: boolean;
+}
+
+export default class InteractiveTerminal extends React.Component<
+  InteractiveTerminalProps,
+  InteractiveTerminalState
+> {
```
Comment (dotcs) [inline src/components/hero/interactive.tsx:26]: I guess this could also be written as a FunctionalComponent. But I guess it's also fine to keep is as a class since it's quite complex.

File: src/components/hero/subcomponents.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+interface SubmitButtonProps {
+  loading: boolean;
+}
+
+export const SubmitButton: FC<SubmitButtonProps> = (props) => (
+  <button
+    type="submit"
+    className="inline-flex items-center ml-10 mb-10 float-left bg-tsfresh-green-500 hover:bg-green-700 text-white font-bold py-2 px-4 rounded"
+  >
+    {props.loading ? <Spinner /> : null} Submit
```
Comment (dotcs) [inline src/components/hero/subcomponents.tsx]: I would put the spinner after the text to avoid "jumping" text, but it's a matter of taste I guess.

File: src/components/hero/interactive.tsx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+  componentWillMount() {
+    this.setState({
+      column_id: "",
+      column_value: "",
+      column_sort: "",
+      column_kind: "",
+      data_file: "",
+      loading: false,
+    });
+  }
```
Comment (dotcs) [inline src/components/hero/interactive.tsx]: You can directly set `state` as a class member instead of using the `componentWillMount` callback.

File: src/components/hero/interactive.tsx [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+    }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const rawData = [...formData.entries()];
+    let params = new URLSearchParams(
+      rawData.filter((x) => x[1] != "" && x[0] != "data_file")
```
Comment (dotcs) [inline src/components/hero/interactive.tsx]:
```suggestion
      rawData.filter((x) => x[1] !== "" && x[0] !== "data_file")
```

But I would consider renaming `x` since it's kind of unreadable this way. What about `filter(([value_name_1, value_name_2]) => ...)` with better names for `value_name_1` and `value_name_2`.

File: src/components/hero/subcomponents.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+const WithLabel: FC<WithLabelProps> = (props) => (
+  <>
+    <span className="text-blue-600">"</span>
+    <label className="w-5 py-2 bg-black text-blue-600 rounded-lg shadow-lg">
+      {props.children}
+    </label>
+    <span className="text-blue-600">"</span>
+  </>
+);
```
Comment (dotcs) [inline src/components/hero/subcomponents.tsx]: I would suggest to put those components into separate files. One component per file ;)

**Review by dotcs — COMMENTED** [review]
LGTM. Well done @nils-braun! A few small remarks, but in general it's great. :+1:

File: src/components/hero/interactive.tsx (hunk 3 above)
Comment (nils-braun) [reply src/components/hero/interactive.tsx]: Yes, definitely. I want to have a indention. But I do not know how to do better

**Comment by nils-braun** [comment]
Thank you very much for your feedback!

**Comment by nils-braun** [comment]
Very nice! I think that made the PR much better!
